### PR TITLE
add aborted and isNavigationFailure

### PR DESCRIPTION
**Pull request by posva**
[Rendered](https://github.com/posva/rfcs/blob/router/amend-navigation-failures/active-rfcs/0033-router-navigation-failures.md)

### CODE REVIEW

File: active-rfcs/0033-router-navigation-failures.md [hunk 1]
```diff
@@ -148,27 +149,49 @@ router.currentRoute // { path: '/login', redirectedFrom: { path: '/profile/dashb
...
+  }
+  // Only duplicated navigations
+  if (isNavigationFailure(failure, NavigationFailureType.duplicated)) {
+    // ...
+  }
+  // Aborted or canceled navigations
+  if (
+    isNavigationFailure(
+      failure,
+      NavigationFailureType.aborted | NavigationFailureType.canceled
```
Comment (kiaking) [inline active-rfcs/0033-router-navigation-failures.md:181]: Nit pick but this should be `NavigationFailureType.aborted || NavigationFailureType.canceled` (double pipe)?

Comment (posva) [reply active-rfcs/0033-router-navigation-failures.md:181]: No, it's a bitwise OR. People can still do

```js
isNavigationFailure(failure, NavigationFailureType.aborted) || isNavigationFailure(failure, NavigationFailureType.aborted)
```

I think we could show this one first and say in a note we can use the bitwise operator as well. I use the bitwise one inside the codebase to make it smaller

Comment (kiaking) [reply active-rfcs/0033-router-navigation-failures.md:181]: Ahhh I see thanks! Good to know 😳

Comment (posva) [reply active-rfcs/0033-router-navigation-failures.md:181]: I unresolved the question in case other people want to provide their input too 🙂

**Comment by posva** [comment]
This RFC is now in final comments stage. An RFC in final comments stage means that:

The core team has reviewed the feedback and reached consensus about the general direction of the RFC and believes that this RFC is a worthwhile addition to the framework.
Final comments stage does not mean the RFC's design details are final - we may still tweak the details as we implement it and discover new technical insights or constraints. It may even be further adjusted based on user feedback after it lands in an alpha/beta release.
If no major objections with solid supporting arguments have been presented after a week, the RFC will be merged and become an active RFC.

**Pull request merged**